### DISCUSSION TITLE
permission for external user on HPC

### DISCUSSION THREAD

**Discussion by mcacace** [idaholab/moose#26024] [Q&A Getting Started]
Dear all,
I am facing some issues while trying to open my moose installation to other users on a HPC cluster (each having their own working directory). My (bad, I know ...) idea is to let them building our moose-based app without (I know it is not the right way, but ... ) requiring them to go through the libmesh installation (I also provide my own petsc installation to them). 
So, now the issue. All works fine if the other users are linking to my installation and "making" the app without loading any external module available in the framework. However, whenever they try to link any additional module (via the makefile input of the app) they are getting errors (i.e. permission denied - see below). 

Is there any way I can free all dependencies to any other user? And, out of curiosity, why is it working if no module are loaded? 

Thanks for any help,
mauro

This is the template of the error they receive --> the user is running inside (/home/bloech) while linking to the framework compiled inside (/home/cacace).
```
Creating Unity /storage/vast-gfz-hpc-01/home/bloech/projects/golem/build/unity_src/userobjects_Unity.C
/home/cacace/projects/moose_petsc_3_20_0/modules/module_loader/src/ModulesApp.C:504:1: f**atal error: opening dependency file /home/cacace/projects/moose_petsc_3_20_0/modules/module_loader/src/ModulesApp_with_fp_ray_rdg_ht_ns.x86_64-pc-linux-gnu.opt.lo.d: Permission denied**
  504 | 
```

**Comment by GiudGiud** [comment]
Hello

What are the permissions for 
```
/home/cacace/projects/moose_petsc_3_20_0/modules/module_loader/src/ModulesApp_with_fp_ray_rdg_ht_ns.x86_64-pc-linux-gnu.opt.lo.d
```
`ls -l` will report on those. 
and could you just set (recursively) open permissions (read at least) for the whole folder?

Guillaume

**Reply by mcacace** [reply]
Thanks I will give it a try and report back.
Mauro

**Comment by mcacace** [comment]
After opening some additional permissions it worked. Thanks!
